### PR TITLE
TextSentimentalAnalysisScipt

### PR DESCRIPTION
**Pull request by shantamsultania**
# Description

A GUI that will allow you to convert perform Sentimental analysis on a particular text entered by the user 

## Fixes #(issue_no)

Fixes: #211

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] Documentation Update

## Checklist:

- [x ] My code follows the style guidelines(Clean Code) of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have created a helpful and easy to understand `README.md`
- [x ] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] I have added the project meta data in the PR template.
- [x ] I have created the ``requirements.txt`` file if needed.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [x] AI/ML
- [ ] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [ ] Games
- [ ] Networking
- [ ] OS_Utilities
- [ ] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [ ] Fun
- [ ] Others

Title: Text Sentimental Analysis

Folder: Text_Sentimental_Analysis_Script_with_GUI

Requirements: none

Script: app.py

Arguments: none

Contributor: shantamsultania 

Description: A GUI that will allow you to convert perform Sentimental analysis on a particular text entered by the user

### CODE REVIEW

**Comment by shantamsultania** [comment]
> _No description provided._

**Comment by shantamsultania** [comment]
@kaustubhgupta @antrikshmisri any suggestion in the same ?

File: Text-Sentimental-Analysis-Scipt-with-GUI/Model.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                      "too", "very", "s", "t", "can", "will", "just", "don", "should", "now"]
+
+        # Removing stop words from the tokenized words list
+        final_words = []
+        for word in tokenized_words:
+            if word not in stop_words:
+                final_words.append(word)
+
+        emotion_list = []
+        with open('textFiles/emotions.txt', 'r') as file:
```
Comment (kaustubhgupta) [inline Text-Sentimental-Analysis-Scipt-with-GUI/Model.py:48]:
```suggestion
        with open('./Text-Sentimental-Analysis-Script-with-GUI/textFiles/emotions.txt', 'r') as file:
```

File: Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+root = Tk()
+
+root.geometry("1280x720+1000+30")
+
+root.title("Text Sentimental Analysis")
+
+root.resizable(False, False)
+
+image = PhotoImage(file="images/background.png")
```
Comment (kaustubhgupta) [inline Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py:30]:
```suggestion
image = PhotoImage(file="./Text-Sentimental-Analysis-Script-with-GUI/images/background.png")
```

File: Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+enterWordLabel = Label(root, text='Enter Text for Analysis', font=('castellar', 15, 'bold'), fg='black')
+enterWordLabel.place(x=180, y=20)
+
+enterWordEntry = Entry(root, font=('arial', 20, 'bold'), bd=5, justify=CENTER)
+enterWordEntry.place(x=100, y=70)
+
+enterWordEntry.focus_set()
+
+searchImage = PhotoImage(file='images/search.png')
```
Comment (kaustubhgupta) [inline Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py:42]:
```suggestion
searchImage = PhotoImage(file='./Text-Sentimental-Analysis-Script-with-GUI/images/search.png')
```

File: Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import textSentimentalAnalysisWebApp.Model
```
Comment (kaustubhgupta) [inline Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py:1]:
```suggestion
import Model
```

File: Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+from tkinter import *
+from tkinter import messagebox
+
+
+def search():
+    word = enterWordEntry.get()
+
+    text = word.strip()
+    model_object = textSentimentalAnalysisWebApp.Model.model()
```
Comment (kaustubhgupta) [inline Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py:11]:
```suggestion
    model_object = Model.model()
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Bugs:
- The folder name has wrong spelling of `Script`,  `Text-Sentimental-Analysis-Scipt-with-GUI/`. Kindly correct that. I am assuming that you have corrected that and based on this correction, I am suggesting these next steps.  

- Add a blank `__init__.py` file inside the Text-Sentimental-Analysis-Script-with-GUI folder so that this becomes package and you can directly call the Model. I have suggested the same in the code.

- For some reason, this simple line is not getting any sentiment and throwing this error:
image

**Comment by shantamsultania** [comment]
hey @kaustubhgupta i will look into this and clear out the changes just stuck with some work so will clear out all the 3 pr changes tomorrow it self

**Comment by kaustubhgupta** [comment]
> hey @kaustubhgupta i will look into this and clear out the changes just stuck with some work so will clear out all the 3 pr changes tomorrow it self

Are you facing any issues as you haven't updated any of the 3 PRs?

**Comment by shantamsultania** [comment]
The thing is that i am busy with some work here in office will be done by today so will make changes after that

**Comment by kaustubhgupta** [comment]
> The thing is that i am busy with some work here in office will be done by today so will make changes after that

I can give you the last 2-day extension to fix all 3 PRs and after that, I will be forced to close the PRs and issues.

**Comment by shantamsultania** [comment]
No issue i have made the changes just need to update the PR will do that in
evening so that you can review them thanks for the extension

File: Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py (hunk 5 above)
Comment (shantamsultania) [reply Text-Sentimental-Analysis-Scipt-with-GUI/text_sentimental_application.py]: please if all the suggested issues are resolved or not let me know if any more changes are required

File: Text_Sentimental_Analysis_Script_with_GUI/app.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+from tkinter import *
+from tkinter import messagebox
+from Text_Sentimental_Analysis_Script_with_GUI import Model
```
Comment (kaustubhgupta) [inline Text_Sentimental_Analysis_Script_with_GUI/app.py:3]:
```suggestion
import Model
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Just a minor change:

**Comment by shantamsultania** [comment]
done the minor change

**Review by kaustubhgupta — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Comment by shantamsultania** [comment]
> @shantamsultania Add the requirements.txt file and update the PR template with project metadata. Look at this PR for reference: #1002

please check the requested changes as tkinter is preinstalled I don't think it requires any requirements.txt file

**Comment by kaustubhgupta** [comment]
> > @shantamsultania Add the requirements.txt file and update the PR template with project metadata. Look at this PR for reference: #1002
> 
> please check the requested changes as tkinter is preinstalled I don't think it requires any requirements.txt file

oh, got it

**Comment by kaustubhgupta** [comment]
@shantamsultania The PR template has been updated wrongly. Refer to approved PRs or the one I pointed in the above message. The category is missing.

**Comment by shantamsultania** [comment]
hey @kaustubhgupta i have updated the template please check now

**Review by kaustubhgupta — APPROVED** [review]

**Pull request merged**